### ISSUE TITLE
Inconsistent tagging of #TextValue,  #Date and #Duration

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#973]
In below similar text examples, the POS tags for `six` is often `#Value` and `#TextValue`, _but not always_
and `month(s)` is `#Date` and `#Duration` _but not always_.

```javascript
const nlp = require('compromise')

const text = [
  'a six month session',
  'I have agreed a six month session',
  'six months',
  `A six-month session`,
  'I have agreed a six-month session',
]

text.forEach((t) => nlp(t).debug())
```

Outputs:

```ascii
  ┌─────────
  │ 'a'        - Determiner
  │ 'six'      - Value, TextValue, Cardinal
  │ 'month'    - Date, Noun, Duration
  │ 'session'  - Noun, Singular



  ┌─────────
  │ 'six'      - Value, TextValue, Cardinal
  │ 'months'   - Noun, Plural



  ┌─────────
  │ 'A'        - Determiner
  │ 'six'      - Value, TextValue, Cardinal
  │ 'month'    - Date, Noun, Duration
  │ 'session'  - Noun, Singular



  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'have'     - Verb, Auxiliary
  │ 'agreed'   - Verb, PastTense
  │ 'a'        - Determiner
  │ 'six'      - Noun, Singular
  │ 'month'    - Date, Noun, Duration
  │ 'session'  - Noun, Singular




  ┌─────────
  │ 'six'      - Value, TextValue, Cardinal
  │ 'months'   - Noun, Plural



  ┌─────────
  │ 'A'        - Determiner
  │ 'six'      - Value, TextValue, Cardinal
  │ 'month'    - Date, Noun, Duration
  │ 'session'  - Noun, Singular



  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'have'     - Verb, Auxiliary
  │ 'agreed'   - Verb, PastTense
  │ 'a'        - Determiner
  │ 'six'      - Noun, Singular
  │ 'month'    - Date, Noun, Duration
  │ 'session'  - Noun, Singular
```

It seems like these should have a more consistent tagging output?
I can take a look at improving this in a PR if you give me some pointers on where to configure?

**Comment by thegoatherder** [comment]
It's worth pointing out that in Compromise `v13.11.4` the output is different - all occurrences of `six` are recognised as `#TextValue` and `#Value` - so this appears to be a regression in `v14`

```ascii
=====
  -----
  ｜ 'a'        - Value
  ｜ 'six'      - TextValue, Value, Cardinal
  ｜ 'month'    - Noun, Singular
  ｜ 'session'  - Noun, Singular

=====
  -----
  ｜ 'I'        - Pronoun, Noun, Singular
  ｜ 'have'     - Infinitive, PresentTense, Verb, Auxiliary
  ｜ 'agreed'   - PastTense, Verb
  ｜ 'a'        - Value
  ｜ 'six'      - TextValue, Value, Cardinal, Singular, Noun
  ｜ 'month'    - Noun, Singular
  ｜ 'session'  - Noun, Singular

=====
  -----
  ｜ 'six'      - TextValue, Value, Cardinal
  ｜ 'months'   - Noun, Plural

=====
  -----
  ｜ 'A'        - Value
  ｜ 'six'      - TextValue, Value, Cardinal
  ｜ 'month'    - Noun, Singular
  ｜ 'session'  - Noun, Singular

=====
  -----
  ｜ 'I'        - Pronoun, Noun, Singular
  ｜ 'have'     - Infinitive, PresentTense, Verb, Auxiliary
  ｜ 'agreed'   - PastTense, Verb
  ｜ 'a'        - Value
  ｜ 'six'      - TextValue, Value, Cardinal, Singular, Noun
  ｜ 'month'    - Noun, Singular
  ｜ 'session'  - Noun, Singular
```

**Comment by spencermountain** [comment]
hey, yeah to debug the tagger:
```js
nlp.verbose('tagger')
nlp('I have agreed a six month session').debug()
```
you can see it becomes a Noun in post, like `did a 900, paid a 20`. Probably just a sloppy rule we can remove.

you can also run `npm run match` and wait a minute, then search for 'six' not tagged as a number `(six && !#TextValue)`
prs welcomed
cheers

**Comment by thegoatherder** [comment]
@spencermountain this is pretty much closed as it mostly duplicates #972 and #992
There is still an issue in `Compromise 14.8.2` with the term `months` not getting tagged as `#Duration`. I think it should be? I have patched on my end via plugin. We tag `(weeks|months|years|days)` as `#Duration`. 

I will do a PR for this.
